### PR TITLE
[cisco_ios] fix timestamp with zones parsing

### PR DESCRIPTION
**Pull request by pkoutsovasilis**
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

N/A

## How to test this PR locally

`elastic-package test pipeline -v`

## Related issues

- Related https://github.com/elastic/integrations/issues/9857

## Screenshots

image

Kibana displays the timestamp in the timezone of the client
image

### CODE REVIEW

**Comment by pkoutsovasilis** [comment]
@taylor-swanson can you have a look at it? It is peculiar but I think it gets the job with the timestamps and zone done

File: packages/cisco_ios/data_stream/log/_dev/test/pipeline/test-syslog.log-expected.json [hunk 1]
```diff
@@ -582,7 +624,7 @@
                 "provider": "firewall",
                 "sequence": 1469087,
                 "severity": 5,
-                "timezone": "UTC",
+                "timezone": "CDT",
```
Comment (taylor-swanson) [inline packages/cisco_ios/data_stream/log/_dev/test/pipeline/test-syslog.log-expected.json:627]: Timestamp for this event still appears to be incorrect.

original: `Mar 29 07:40:10.863 CDT`
`@timestamp`: `2024-03-29T07:40:10.863Z`

Comment (pkoutsovasilis) [reply packages/cisco_ios/data_stream/log/_dev/test/pipeline/test-syslog.log-expected.json:627]: please give it another look @taylor-swanson

File: packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -109,7 +133,8 @@ processors:
         - "MMM d HH:mm:ss.SSS"
         - "MMM d HH:mm:ss z"
         - "MMM d HH:mm:ss"
-      timezone: '{{{event.timezone}}}'
+      output_format: "yyyy-MM-dd'T'HH:mm:ss.SSSXXX"
```
Comment (taylor-swanson) [inline packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:136]: Is this the default format? If so, we can remove it.

Comment (pkoutsovasilis) [reply packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml:136]: it is but there was a point I was reverse-engineering this so I tried to mess with all knobs 😆

**Review by taylor-swanson — APPROVED** [review]

**Pull request merged**